pkg/conversion/capi2mapi/powervs.go

```diff
@@ -0,0 +1,305 @@
+	if m.machineSet == nil || m.template == nil || m.powerVSCluster == nil || m.machineAndPowerVSMachineAndPowerVSCluster == nil {
+		return nil, nil, errCAPIMachineSetPowerVSMachineTemplatePowerVSClusterCannotBeNil
+	}
+
+	var (
+		errors   []error
+		warnings []string
+	)
+
+	// Run the full ToMachine conversion so that we can check for
+	// any Machine level conversion errors in the spec translation.
+	mapiPowerVSMachine, warn, err := m.ToMachine()
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	warnings = append(warnings, warn...)
+
+	mapiMachineSet, err := fromCAPIMachineSetToMAPIMachineSet(m.machineSet)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	mapiMachineSet.Spec.Template.Spec = mapiPowerVSMachine.Spec
+
+	// Copy the labels and annotations from the Machine to the template.
+	mapiMachineSet.Spec.Template.ObjectMeta.Annotations = mapiPowerVSMachine.ObjectMeta.Annotations
+	mapiMachineSet.Spec.Template.ObjectMeta.Labels = mapiPowerVSMachine.ObjectMeta.Labels
+
+	if len(errors) > 0 {
+		return nil, warnings, utilerrors.NewAggregate(errors)
+	}
+
+	return mapiMachineSet, warnings, nil
+}
+
+// toProviderSpec converts a capi2mapi machineAndPowerVSMachineAndPowerVSCluster into a MAPI PowerVSMachineProviderConfig.
+//
+//nolint:funlen
+func (m machineAndPowerVSMachineAndPowerVSCluster) toProviderSpec() (*mapiv1.PowerVSMachineProviderConfig, []string, error) {
+	if m.machine == nil || m.powerVSMachine == nil || m.powerVSCluster == nil {
+		return nil, nil, errCAPIMachinePowerVSMachinePowerVSClusterCannotBeNil
+	}
+
+	var (
+		warnings []string
+		errors   field.ErrorList
+	)
+	fldPath := field.NewPath("spec")
+	serviceInstance, err := convertPowerVSServiceInstanceToMAPI(fldPath.Child("serviceInstance"), m.powerVSMachine.Spec.ServiceInstanceID, m.powerVSMachine.Spec.ServiceInstance)
+	if err != nil {
+		errors = append(errors, err)
+	}
+	image, err := convertPowerVSImageToMAPI(fldPath.Child("image"), m.powerVSMachine.Spec.Image, m.powerVSMachine.Spec.ImageRef)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	network, err := convertPowerVSNetworkToMAPI(fldPath.Child("network"), m.powerVSMachine.Spec.Network)
+	if err != nil {
+		errors = append(errors, err)
+	}
+
+	mapiProviderConfig := mapiv1.PowerVSMachineProviderConfig{
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "PowerVSMachineProviderConfig",
+			APIVersion: "machine.openshift.io/v1",
+		},
+		ServiceInstance: serviceInstance,
+		Image:           image,
+		Network:         network,
+		KeyPairName:     m.powerVSMachine.Spec.SSHKey,
+		SystemType:      m.powerVSMachine.Spec.SystemType,
+		ProcessorType:   mapiv1.PowerVSProcessorType(m.powerVSMachine.Spec.ProcessorType),
+		Processors:      m.powerVSMachine.Spec.Processors,
+		MemoryGiB:       m.powerVSMachine.Spec.MemoryGiB,
+		//CredentialsSecret:
+		//LoadBalancers: TODO: Not supported for workers.
```

Comment:
Are you creating Jira trackers for the feature gaps you've found so far?

Reply:
Yeah, I will do

Reply:
Created the task https://issues.redhat.com/browse/MULTIARCH-5041